src/o365/graph/commands/teams/teams-channel-add.ts

```diff
@@ -0,0 +1,149 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import GraphCommand from "../../GraphCommand";
+import * as request from 'request-promise-native';
+import Utils from '../../../../Utils';
+import { Channel } from './Channel';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  groupId : string;
+  name: string;
+  description?: string;
+}
+
+class TeamsChannelAddCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.TEAMS_CHANNEL_ADD}`;
+  }
+
+  public get description(): string {
+    return 'Add a channel to a Microsoft Teams team';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.groupId = args.options.groupId;
+    telemetryProps.displayName = args.options.name;
+    telemetryProps.description = args.options.description;
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+
+    let endpoint: string = `${auth.service.resource}/beta/teams/${args.options.groupId}/channels`;
+   
+    auth
+        .ensureAccessToken(auth.service.resource, cmd, this.debug)
+        .then((): request.RequestPromise => {
+          const requestOptions: any = {
+            url: endpoint,
+            headers: Utils.getRequestHeaders({
+              authorization: `Bearer ${auth.service.accessToken}`,
+              accept: 'application/json;odata.metadata=none',
+              'content-type': 'application/json;odata=nometadata'
+            }),
+            body: {
+              displayName: args.options.name,
+              description: (args.options.description) ? args.options.description : null
+            },
+            json: true
+          };
+
+          if (this.debug) {
+            cmd.log('Executing web request...');
+            cmd.log(requestOptions);
+            cmd.log('');
+          }
+
+          return request.post(requestOptions);
+        })
+        .then((res: any): void => {
+          var channel : Channel = {
```

Comment:
The raw response already matches this object so there is no need to re-assign it.